internal/controller/reinstall/helper.go

```diff
@@ -220,3 +222,105 @@ func logAndWrapUpdateFailure(log *zap.Logger, currentError, updateError error, e
 	}
 	return fmt.Errorf("%s, update error: %w, error: %w", errorMsg, updateError, currentError)
 }
+
+func setOrUpdateLabel(objectMeta *metav1.ObjectMeta, key, value string) bool {
+	if objectMeta.Labels == nil {
+		objectMeta.Labels = make(map[string]string)
+	}
+	if currentValue, ok := objectMeta.Labels[key]; !ok || currentValue != value {
+		objectMeta.Labels[key] = value
+		return true
+	}
+	return false
+}
+
+func getDataPreservationSummary(
```

Comment:
I will still maintain that this is a strange way to handle this.

Why is counting the resources after the fact better than failing and setting the condition when you're actually doing the backup/restore work?

Reply:
Performing the resource count after the backup/restore operation serves as an additional verification step, ensuring that the intended resources were successfully processed. This post-operation validation helps detect any discrepancies, such as partial failures or unexpected resource states, that may not be immediately evident during the backup/restore execution.

Additionally, separating the counting logic from the backup/restore operation improves function cohesion and maintainability. It keeps the backup/restore logic focused on execution while delegating validation to a distinct step, making both processes more modular, testable, and reusable.

Reply:
I guess my question was more about how important the actual number is.

I'm fine with validating, I'm fine with splitting validation, but I don't think anyone is ever going to do anything with the specific number of resources in each of these buckets. It just feels like noise to me.

Reply:
Fair point regarding the utility of the preserved & restored resource counts. I will create a separate PR to replace these counts with information as to which resources were preserved & restored- and then perform an added validation on the restored resources (ensuring they account for the preserved versions).